system-tests/lib/dep-installer/index.ts

```diff
@@ -0,0 +1,310 @@
+ */
+async function makeWorkspacePackagesAbsolute (pathToPkgJson: string): Promise<string[]> {
+  const pkgJson = await fs.readJson(pathToPkgJson)
+  const updatedDeps: string[] = []
+
+  for (const deps of [pkgJson.dependencies, pkgJson.devDependencies, pkgJson.optionalDependencies]) {
+    for (const dep in deps) {
+      const version = deps[dep]
+
+      if (version.startsWith('file:')) {
+        const absPath = pathToPackage(dep)
+
+        log(`Setting absolute path in package.json for ${dep}: ${absPath}.`)
+
+        deps[dep] = `file:${absPath}`
+        updatedDeps.push(dep)
+      }
+    }
+  }
+
+  await fs.writeJson(pathToPkgJson, pkgJson)
+
+  return updatedDeps
+}
+
+/**
+ * Given a `system-tests` project name, detect and install the `node_modules`
+ * specified in the project's `package.json`. No-op if no `package.json` is found.
+ * Will use `yarn` or `npm` based on the lockfile present.
+ */
+export async function scaffoldProjectNodeModules (project: string, updateLockFile: boolean = !!process.env.UPDATE_LOCK_FILE): Promise<void> {
+  const projectDir = projectPath(project)
+  const relativePathToMonorepoRoot = path.relative(
+    path.join(projects, project),
+    path.join(root, '..'),
+  )
+  const projectPkgJsonPath = path.join(projectDir, 'package.json')
+
+  const runCmd = async (cmd) => {
+    log(`Running "${cmd}" in ${projectDir}`)
+    await execa(cmd, { cwd: projectDir, stdio: 'inherit', shell: true })
+  }
+
+  const cacheNodeModulesDir = path.join(cachedir('cy-system-tests-node-modules'), project, 'node_modules')
+  const tmpNodeModulesDir = path.join(projectPath(project), 'node_modules')
+
+  async function removeWorkspacePackages (packages: string[]): Promise<void> {
+    for (const dep of packages) {
+      const depDir = path.join(tmpNodeModulesDir, dep)
+
+      log('Removing', depDir)
+      await fs.remove(depDir)
+    }
+  }
+
+  try {
+    // this will throw and exit early if the package.json does not exist
+    const pkgJson: SystemTestPkgJson = require(projectPkgJsonPath)
+
+    log(`Found package.json for project ${project}.`)
+
+    if (pkgJson._cySkipDepInstall) {
+      return log(`_cySkipDepInstall set in package.json, skipping dep-installer steps`)
+    }
+
+    if (!pkgJson.dependencies && !pkgJson.devDependencies && !pkgJson.optionalDependencies) {
+      return log(`No dependencies found, skipping dep-installer steps`)
+    }
+
+    const lockFilename = await getLockFilename(projectDir)
+    const hasYarnLock = lockFilename === 'yarn.lock'
+
+    // 1. Ensure there is a cache directory set up for this test project's `node_modules`.
+    await ensureCacheDir(cacheNodeModulesDir)
+
+    let persistCacheCb: () => Promise<void>
+
+    if (hasYarnLock) {
+      await symlinkNodeModulesFromCache(tmpNodeModulesDir, cacheNodeModulesDir)
+    } else {
+      // due to an issue in NPM, we cannot have `node_modules` be a symlink. fall back to copying.
+      // https://github.com/npm/npm/issues/10013
```

Comment:
Seems the linked issue is 3-4 years old - I wonder if/when this will be fixed, or if it's a wontfix. :cry: